scripts/languageserver/parse.jl

```diff
@@ -0,0 +1,243 @@
+type VarInfo
+    t
+    doc::String
+end
+
+type Block
+    uptodate::Bool
+    ex::Any
+    range::Range
+    name::String
+    var::VarInfo
+    localvar::Dict{String,VarInfo}
+    diags::Vector{Diagnostic}
+end
+
+function Block(utd, ex, r::Range)
+    t, name, doc, lvars = classify_expr(ex)
+    ctx = LintContext()
+    ctx.lineabs = r.start.line+1
+    dl = r.end.line-r.start.line-ctx.line
+    Lint.lintexpr(ex, ctx)
+    diags = map(ctx.messages) do l
+        return Diagnostic(Range(Position(r.start.line+l.line+dl-1, 0), Position(r.start.line+l.line+dl-1, 100)),
+                        LintSeverity[string(l.code)[1]],
+                        string(l.code),
+                        "Lint.jl",
+                        l.message) 
+    end
+    v = VarInfo(t, doc)
+
+    return Block(utd, ex, r, name,v, lvars, diags)
+end
+
+function Base.parse(uri::String, server::LanguageServer, updateall=false)
```

Comment:
I think we should add a method to ``Base.parse`` here, because ``Base.parse`` is assumed to return something. So maybe just find some other name for this?